lib/Basics/ResourceUsage.h

```diff
@@ -146,7 +146,7 @@ class ResourceUsageAllocatorBase : public Allocator {
   using size_type = typename std::allocator_traits<Allocator>::size_type;
   using value_type = typename std::allocator_traits<Allocator>::value_type;
 
-  ResourceUsageAllocatorBase() = delete;
+  ResourceUsageAllocatorBase() = default;
```

Comment:
is this change still necessary?

Reply:
Yes right now it unfortunately is. The compiler complains if not set to default: 

```
/home/hkernbach/Git/ArangoDB/devel-3/tests/Graph/TraverserCacheTest.cpp:58:27: error: call to implicitly-deleted default constructor of 'arangodb::MonitoredCollectionToShardMap' (aka 'unordered_map<basic_string<char, std::char_traits<char>, ResourceUsageAllocator<char, arangodb::ResourceMonitor>>, vector<basic_string<char, std::char_traits<char>, ResourceUsageAllocator<char, arangodb::ResourceMonitor>>, ResourceUsageAllocator<basic_string<char, std::char_traits<char>, ResourceUsageAllocator<char, arangodb::ResourceMonitor>>, arangodb::ResourceMonitor>>, arangodb::hash_monitored_string, arangodb::compare_monitored_string, ResourceUsageAllocator<pair<const basic_string<char, std::char_traits<char>, ResourceUsageAllocator<char, arangodb::ResourceMonitor>>, vector<basic_string<char, std::char_traits<char>, ResourceUsageAllocator<char, arangodb::ResourceMonitor>>, ResourceUsageAllocator<basic_string<char, std::char_traits<char>, ResourceUsageAllocator<char, arangodb::ResourceMonitor>>, arangodb::ResourceMonitor>>>, arangodb::ResourceMonitor>>')
      collectionToShardMap{};  // can be empty, only used in standalone mode
                          ^~
/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/unordered_map.h:141:7: note: explicitly defaulted function was implicitly deleted here
      unordered_map() = default;
      ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/unordered_map.h:105:18: note: default constructor of 'unordered_map<std::basic_string<char, std::char_traits<char>, arangodb::ResourceUsageAllocator<char, arangodb::ResourceMonitor>>, std::vector<std::basic_string<char, std::char_traits<char>, arangodb::ResourceUsageAllocator<char, arangodb::ResourceMonitor>>, arangodb::ResourceUsageAllocator<std::basic_string<char, std::char_traits<char>, arangodb::ResourceUsageAllocator<char, arangodb::ResourceMonitor>>, arangodb::ResourceMonitor>>, arangodb::hash_monitored_string, arangodb::compare_monitored_string, arangodb::ResourceUsageAllocator<std::pair<const std::basic_string<char, std::char_traits<char>, arangodb::ResourceUsageAllocator<char, arangodb::ResourceMonitor>>, std::vector<std::basic_string<char, std::char_traits<char>, arangodb::ResourceUsageAllocator<char, arangodb::ResourceMonitor>>, arangodb::ResourceUsageAllocator<std::basic_string<char, std::char_traits<char>, arangodb::ResourceUsageAllocator<char, arangodb::ResourceMonitor>>, arangodb::ResourceMonitor>>>, arangodb::ResourceMonitor>>' is implicitly deleted because field '_M_h' has a deleted default constructor
      _Hashtable _M_h;
                 ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/hashtable.h:531:7: note: explicitly defaulted function was implicitly deleted here
      _Hashtable() = default;
      ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/hashtable.h:201:7: note: default constructor of '_Hashtable<std::basic_string<char, std::char_traits<char>, arangodb::ResourceUsageAllocator<char, arangodb::ResourceMonitor>>, std::pair<const std::basic_string<char, std::char_traits<char>, arangodb::ResourceUsageAllocator<char, arangodb::ResourceMonitor>>, std::vector<std::basic_string<char, std::char_traits<char>, arangodb::ResourceUsageAllocator<char, arangodb::ResourceMonitor>>, arangodb::ResourceUsageAllocator<std::basic_string<char, std::char_traits<char>, arangodb::ResourceUsageAllocator<char, arangodb::ResourceMonitor>>, arangodb::ResourceMonitor>>>, arangodb::ResourceUsageAllocator<std::pair<const std::basic_string<char, std::char_traits<char>, arangodb::ResourceUsageAllocator<char, arangodb::ResourceMonitor>>, std::vector<std::basic_string<char, std::char_traits<char>, arangodb::ResourceUsageAllocator<char, arangodb::ResourceMonitor>>, arangodb::ResourceUsageAllocator<std::basic_string<char, std::char_traits<char>, arangodb::ResourceUsageAllocator<char, arangodb::ResourceMonitor>>, arangodb::ResourceMonitor>>>, arangodb::ResourceMonitor>, std::__detail::_Select1st, arangodb::compare_monitored_string, arangodb::hash_monitored_string, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Prime_rehash_policy, std::__detail::_Hashtable_traits<true, false, true>>' is implicitly deleted because base class '_Hashtable_enable_default_ctor<arangodb::compare_monitored_string, arangodb::hash_monitored_string, arangodb::ResourceUsageAllocator<std::pair<const std::basic_string<char, std::char_traits<char>, arangodb::ResourceUsageAllocator<char, arangodb::ResourceMonitor>>, std::vector<std::basic_string<char, std::char_traits<char>, arangodb::ResourceUsageAllocator<char, arangodb::ResourceMonitor>>, arangodb::ResourceUsageAllocator<std::basic_string<char, std::char_traits<char>, arangodb::ResourceUsageAllocator<char, arangodb::ResourceMonitor>>, arangodb::ResourceMonitor>>>, arangodb::ResourceMonitor>>' (aka '_Enable_default_constructor<__and_<is_default_constructible<arangodb::compare_monitored_string>, is_default_constructible<arangodb::hash_monitored_string>, is_default_constructible<arangodb::ResourceUsageAllocator<std::pair<const std::basic_string<char, std::char_traits<char>, arangodb::ResourceUsageAllocator<char, arangodb::ResourceMonitor>>, std::vector<std::basic_string<char, std::char_traits<char>, arangodb::ResourceUsageAllocator<char, arangodb::ResourceMonitor>>, arangodb::ResourceUsageAllocator<std::basic_string<char, std::char_traits<char>, arangodb::ResourceUsageAllocator<char, arangodb::ResourceMonitor>>, arangodb::ResourceMonitor>>>, arangodb::ResourceMonitor>>>{}, __detail::_Hash_node_base>') has a deleted default constructor
      private _Hashtable_enable_default_ctor<_Equal, _Hash, _Alloc>
      ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/enable_special_members.h:113:15: note: '_Enable_default_constructor' has been explicitly marked deleted here
    constexpr _Enable_default_constructor() noexcept = delete;
              ^
1 error generated.
gmake[2]: *** [tests/Graph/CMakeFiles/arango_tests_graph.dir/build.make:90: tests/Graph/CMakeFiles/arango_tests_graph.dir/TraverserCacheTest.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
```

Do you have a better approach here?

Reply:
ah - I know how. On it.

Reply:
We need to provide an allocator from the test, then we can again delete the default constructor.

Reply:
added: 938324ba9ea35224851ff2471843a431aaca15a3